elytron/WFLY-16793-oidc-identity-propagation.adoc

```diff
@@ -0,0 +1,166 @@
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1971[EAP7-1971]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+
+TBD
+
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* It will be possible to propagate an identity from a WAR to an EJB within a single EAR without
+requiring any new configuration to be added to enable this behaviour. This will just automatically
+work.
+
+* A new `virtual-security-domain` resource will be introduced in the `elytron` subsystem.
+The new resource will make it possible to outflow security identities established by virtual
+security domains to other security domains. The new `virtual-security-domain` resource will have
+two attributes:
+
+** `name` - This is the runtime name of a deployment associated with a virtual security domain (e.g.,
+`DEPLOYMENT_NAME.ear`, a deployment that has a subdeployment that is secured using OIDC).
+
+** `outflow-security-domains` - This is the list of `security-domains` that security identities from
+the virtual security domain should be automatically outflowed to. This will be a required attribute
+and must be non-empty.
+
+** As an example, a `virtual-security-domain` could be added as follows:
+[source]
+----
+/subsystem=elytron/virtual-security-domain=DEPLOYMENT_NAME.ear:add(outflow-security-domains=...)
+----
+
+* A new `trusted-virtual-security-domains` attribute will be added to the `security-domain` resource
+in the `elytron` subsystem. This is the list of runtime names of deployments associated with
+virtual security domains that should be trusted by a `security-domain`. This will be an optional
+attribute.
+** e.g., `trusted-virtual-security-domains=DEPLOYMENT_NAME.ear` indicates that this `security-domain`
+should trust the virtual security domain associated with the `DEPLOYMENT_NAME.ear` deployment.
+
+Note that in general, we are following the same concept of security outflow and trust that
+is used when propagating identities between traditional Elytron `security-domains`. Here, we
+are simply extending this concept for the case where we need to propagate an identity from
+a virtual security domain to a traditional `security-domain`.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+Note that in the future, we might want to consider adding other attributes to the `virtual-security-domain`
```

Comment:
Since we've seen recent requests from the community for this, I think we should consider adding an initial set of attributes as part of this RFE. I've updated the Nice-to-Have Requirements section with this list.